Utils/Functions/Filesystem/Extract/script.js

```diff
@@ -1,144 +1,169 @@
-    return this;
-}
+ * CabExtract class
+ */
+// eslint-disable-next-line no-unused-vars
+class CabExtract {
+    constructor() {
+        // do nothing
+    }
 
-/**
-* sets archive
-* @param {string} archive archive which shall be extracted
-* @returns {CabExtract} CabExtract object
-*/
-CabExtract.prototype.archive = function (archive) {
-    this._archive = archive;
-    return this;
-}
+    /**
+     * Sets the setup wizard
+     *
+     * @param {SetupWizard} wizard The setup wizard
+     * @returns {CabExtract} The CabExtract object
+     */
+    wizard(wizard) {
+        this._wizard = wizard;
+        return this;
+    }
 
-/**
-* sets progress message text
-* @param {string} progressMessage progress message
-* @returns {CabExtract} CabExtract object
-*/
-CabExtract.prototype.message = function (progressMessage) {
-    this._progressMessage = progressMessage;
-    return this;
-}
+    /**
+     * Sets the CAB archive
+     *
+     * @param {string} archive The CAB archive which shall be extracted
+     * @returns {CabExtract} The CabExtract object
+     */
+    archive(archive) {
+        this._archive = archive;
+        return this;
+    }
 
-/**
-* sets destination
-* @param {string} destination place where the archive shall be extracted
-* @returns {CabExtract} CabExtract object
-*/
-CabExtract.prototype.to = function (destination) {
-    this._destination = destination;
-    return this;
+    /**
+     * Sets the progress message text
+     *
+     * @param {string} progressMessage The progress message
+     * @returns {CabExtract} The CabExtract object
+     */
+    message(progressMessage) {
+        this._progressMessage = progressMessage;
+        return this;
+    }
+
+    /**
+     * Sets the destination for the extracted archive content
+     *
+     * @param {string} destination The place where the archive shall be extracted
+     * @returns {CabExtract} The CabExtract object
+     */
+    to(destination) {
+        this._destination = destination;
+        return this;
+    }
+
+    /**
+     * Extracts the archive to the previously set destination
+     *
+     * @param {string} args The arguments for the extraction
+     * @returns {void}
+     */
+    extract(args) {
+        if (this._wizard) {
+            let progressMessage = this._progressMessage;
+            if (!progressMessage) {
+                progressMessage = tr("Please wait while {0} is extracted...", this._archive);
+            }
+
+            this._wizard.progressBar(progressMessage);
+        }
+
+        var processArguments;
+        if (args) {
+            processArguments = ["cabextract"].concat(args).concat([this._archive]);
+        } else {
+            processArguments = ["cabextract", this._archive];
+        }
+
+        print("Extracting to: " + this._destination);
+        mkdir(this._destination);
+
+        const process = new ProcessBuilderClass()
+            .command(Java.to(processArguments, "java.lang.String[]"))
```

Comment:
Does this chaining work?

Reply:
According to https://docs.oracle.com/javase/7/docs/api/java/lang/ProcessBuilder.html it should